lib/events.js

```diff
@@ -850,6 +850,39 @@ function listenerCount(type) {
   return 0;
 }
 
+/**
+ * Returns true if the function is register as listener for the event
+ *
+ * @param {string | symbol} type
+ * @param {Function} listener
+ * @returns {boolean}
+ */
+EventEmitter.prototype.hasListener = function hasListener(type, listener) {
```

Comment:
Why adding this when it is possible to use `ee.listeners(type).includes(listener)`? This might be a bit more optimized but I'm not sure if it makes sense. Will it be used in hot paths?

Reply:
I'd say because it's better DX. People will have to concatenate two methods (and thus two context switches).
Moreover, `_listeners` clones the array all the times, which probably is not good for performance.